pkg/dispatchcli/cmd/manage.go

```diff
@@ -0,0 +1,164 @@
+
+	bootstrapUser = ""
+
+	kubeconfigPath = ""
+)
+
+const (
+	bootstrapSecretName = "dispatch-identity-manager-bootstrap"
+)
+
+// NewCmdManage creates a command object for Dispatch "manage" action
+func NewCmdManage(out, errOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     i18n.T(`manage [--enable-bootstrap-mode | --disable-bootstrap-mode]`),
+		Short:   manageLong,
+		Long:    manageLong,
+		Example: manageExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := runManage(out, errOut, cmd, args)
+			CheckErr(err)
+		},
+	}
+	cmd.Flags().BoolVarP(&enableBootstrapModeFlag, "enable-bootstrap-mode", "e", false, "enable Dispatch bootstrap mode")
+	cmd.Flags().BoolVarP(&disableBootstrapModeFlag, "disable-bootstrap-mode", "d", false, "disable Dispatch bootstrap mode")
+	cmd.Flags().StringVar(&bootstrapUser, "bootstrap-user", "", "specify bootstrap user")
+	cmd.Flags().StringVar(&publicKeyPath, "public-key", "", "public key file path for bootstrap user")
+	cmd.Flags().StringVar(&kubeconfigPath, "kubeconfig", "", "customized absolute path to k8s config file (optional)")
+	return cmd
+}
+
+// prepare k8s client throuhg kubeconfig flag or default kubeconfig location
+func prepareK8sClient() (clientset *kubernetes.Clientset, err error) {
+
+	var configPath string
+	var config *restclient.Config
+
+	// create k8s config
+	if kubeconfigPath != "" {
+		configPath, err = homedir.Expand(kubeconfigPath)
+	} else {
+		homeDir, _ := homedir.Dir()
+		configPath = path.Join(homeDir, ".kube", "config")
+	}
+	config, err = clientcmd.BuildConfigFromFlags("", configPath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to load k8s config: %s", err.Error())
+	}
+
+	// create k8s clientset
+	clientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create k8s client: %s", err.Error())
+	}
+	return clientset, nil
+}
+
+func updateBootstrapSecret(secret *v1.Secret, client *kubernetes.Clientset) error {
+	var err error
+	existed, err := client.CoreV1().Secrets("dispatch").Get(bootstrapSecretName, metav1.GetOptions{})
+	if existed.Name == bootstrapSecretName && err == nil {
+		_, err = client.CoreV1().Secrets("dispatch").Update(secret)
+	} else {
+		_, err = client.CoreV1().Secrets("dispatch").Create(secret)
+	}
+	return err
+}
+
+func runManage(out, errOut io.Writer, cmd *cobra.Command, args []string) error {
+
+	if enableBootstrapModeFlag == disableBootstrapModeFlag {
+		runHelp(cmd, []string{})
+		return fmt.Errorf("Please specify one of options: --enable-bootstrap-mode or --disable-bootstrap-mode")
+	}
+
+	// get k8s client
+	client, err := prepareK8sClient()
+	if err != nil {
+		return err
+	}
+
+	if enableBootstrapModeFlag {
+
+		// get bootstrap user and publicKey
+		if bootstrapUser == "" {
+			return fmt.Errorf("Bootstrap user not found, please provide bootstrap user using --bootstrap-user [BOOTSTRAP_USER]")
+		}
+		publicKeyBytes, err := ioutil.ReadFile(publicKeyPath)
```

Comment:
`--bootstrap-public-key` is optional, only used for jwt based authentication.  Can we make it optional?

Reply:
sure, fixing